app/gui/view/graph-editor/src/builtin/visualization/java_script/scatterPlot.js

```diff
@@ -58,6 +61,14 @@ class ScatterPlot extends Visualization {
         this.points = { labels: VISIBLE_POINTS }
     }
 
+    updatePreprocessor() {
+        let fn = 'x -> process_to_json_text x limit=' + this.limit
```

Comment:
CCing @4e6: Calling `this.setPreprocessor` repeatedly with different code snippet seems to work, but ideally we shall have a single snippet and just invoke it with different `limit` and `bounds` argument. I hope it will be possible to do something like that with _hidden modules_.